olive/workflows/run/run.py

```diff
@@ -182,16 +152,27 @@ def run_engine(config: RunConfig, data_root: str = None):
     if config.azureml_client:
         config.engine.azureml_client_config = config.azureml_client
 
-    # engine
     engine = config.engine.create_engine()
+    accelerator_specs = create_accelerators(engine.target, config.engine.execution_providers)
+
+    auto_optimizer = None
+    run_rls = {}
+    for accelerator_spec in accelerator_specs:
+        passes = config.passes or {}
+        pass_flows = config.pass_flows or []
+        if not config.passes and not config.auto_optimizer_config.disable_auto_optimizer:
+            auto_optimizer = AutoOptimizer(
+                input_model,
+                engine.evaluator_config,
+                accelerator_spec,
+                config.auto_optimizer_config,
+                config.data_configs,
+            )
+            _passes, pass_flows = auto_optimizer.suggest()
```

Comment:
how about the following logic:
```python
pass_map = {}
if if not config.passes and not config.auto_optimizer_config.disable_auto_optimizer:
    for accelerator_spec in accelerator_specs:
        pass_map[[accelerato_spec]] = auto_optimizer.suggest()
else:
    pass_map[accelerator_specs] = (passes, pass_flows)

for accelerator_specs, (passes, pass_flows) in pass_map.items():
    engine.run(xxx)
```

Reply:
In this way, we could decouple the pass generation and engine run. And it is easy to test both logic by using UT easily.

Reply:
Updated.